### PR TITLE
Backport for release of azure-cosmos_4.48.3-hotfix

### PR DESCRIPTION
**Pull request by jeet1995**
Changes summary:
- Following PRs have been cherry-picked:
  - #39077 
  - #37015

This along with removing `System.exit()` calls from `ImplementationBridgeHelpers`.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 ## Release History
 
+### 4.49.0-beta.1 (Unreleased)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:3]: will also need to update the azure-cosmos sdk version in pom.xml to 4.49.0-beta.1, else the pipeline will fail

and then you will need to also update all the versions mentioned in 
`{x-version-update;com.azure:azure-cosmos;dependency} `
`{x-version-update;com.azure:azure-cosmos;current}`

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:3]: you should eb able to use azure-sdk-for-java\eng\versioning update_versions.py if you don't want to do it manually

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: Ran the prepare release script for this to update required files to use 4.48.3-hotfix.

**Comment by jeet1995** [comment]
Observations from the live tests pipeline:
- `ReadTimeoutException` to the gateway in test `createItem_withBulkAndThroughputControlAsNonDefaultGroup` - seems unrelated to this PR.
- `clientTelemetryWithStageJunoEndpoint` is failing - this has been fix in a later PR - unrelated to this PR.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Pull request merged**